docs/contributing/999-rfcs/2022-12-08-wing-inputs.md

```diff
@@ -0,0 +1,202 @@
+---
+title: "#2726 Wing Inputs"
+description: "How to create, share, manage inputs"
+---
+
+# Wing Inputs
+- **Author(s)**: @Hasanaburayyan
+- **Submission Date**: {2023-12-08}
+- **Stage**: Draft
+
+## Background
+
+An input is a value used by either an application or platform that is not necessarily defined within the application's source code.
+
+Inputs can be Platform agnostic, such as an a retry count that is dictated by business requirements. Other inputs may be Platform specific, like a DNS configurations.
+
+### Context
+
+There are a few things that are considered out-of-scope for this RFC. 
+- Inputs are all treated as string data, but helper methods can convert them to primitive Wing types like num, bool, and Json
+- Inputs cannot be created, or updated inflight
+
+## Use cases
+
+The following table contains some use cases and is by no means exhaustive. 
+
+| **Use case** | **Secure** | Consumed |
+| ---- | ---- | ---- |
+| Connection strings | Yes | Inflight |
+| SSH Keys | Yes | Inflight |
+| Bucket count | No | Both |
+| Connection strings | Yes | Inflight |
+| SSH Keys | Yes | Inflight |
+| Bucket count | No | Both |
+| Retry counts | No | Inflight |
+| API Keys | Yes | Inflight |
+| DNS information (certificates, hosted zones) | No | Preflight |
+| Autoscaling parameters | No | Preflight |
+| Feature Flags | No | Both |
+| Platform Input (state-file name, etc..) | No | Preflight |
+
+
+## Configuring and Managing Inputs
+
+### The Wing Configuration File
+
+Inputs are defined via a Wing configuration file. These configuration files are called `wing.toml` and support other file formats such as `json` and `yaml`. 
+
+For instance in order to create a single input that will be used to determine how many buckets are created in our preflight code, we can do the following:
+
+```toml
+# wing.toml
+bucketCount = 3
+```
+
+```js
+// main.w
+bring cloud;
```

Comment:
Is it necessary to include and relate the `Input` philosophy with the `cloud` module? Might be helpful to read inputs from other modules, as in for example `fs` or maybe `http`

Reply:
Yea I considered this a bit and went back and forth on maybe its just a language keyword `input` or some other line of reasoning. 

Here is where my reasoning landed, a cloud.Input is an input that can be used by your application while its running in the cloud, and a key distinguishing factor is these inputs become some cloud resource I.E. an email list in AWS may be stored as an SSM-Parameter which allows operators of the application to go in and alter the input value before the next invocation of some service.

Reply:
Interesting, looking it like that makes sense the name; thanks for the clarification 👍

Reply:
I am still not sure this makes sense under `cloud`. Did you consider `std.Input`?

Not that critical at the moment so I am going with the flow